src/seamless_communication/inference/transcriber.py

```diff
@@ -295,20 +304,42 @@ def transcribe(
             Window size to pad weights tensor.
         :params **sequence_generator_options:
             See BeamSearchSeq2SeqGenerator.
+        :params denoise:
+            Whether to denoise the audio.
+        :params denoise_model:
+            Denoising model to use.
+        :params denoise_two_stems:
+            Whether to use two stems for denoising.
+        :params denoise_float32:
+            Whether to use float32 for denoising.
+        :params denoise_int24:
+            Whether to use int24 for denoising.
 
         :returns:
             - List of Tokens with timestamps.
         """
-        if isinstance(audio, str):
-            with Path(audio).open("rb") as fb:
-                block = MemoryBlock(fb.read())
-            decoded_audio = self.decode_audio(block)
-        else:
-            decoded_audio = {
-                "waveform": audio,
-                "sample_rate": sample_rate,
-                "format": -1,
-            }
+
+        if denoise:
+            demucs = Demucs(
```

Comment:
it would be great to hide that logic into a separate method

Reply:
fixed